apm-collector/apm-collector-worker/src/main/java/org/skywalking/apm/collector/worker/instance/IdentificationSegmentFetcher.java

```diff
@@ -0,0 +1,67 @@
+package org.skywalking.apm.collector.worker.instance;
+
+import java.util.Random;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.skywalking.apm.collector.worker.instance.util.ESLock;
+
+public enum IdentificationSegmentFetcher {
+    INSTANCE;
+
+    private ScheduledFuture task;
+    private Logger logger = LogManager.getLogger(IdentificationSegmentFetcher.class);
+    private ESLock esLock = new ESLock();
+
+    public void fetchSegment(Listener listener) {
+        for (int index = 0; index < 3; index++) {
+            boolean updateSuccess = esLock.tryLock((start, end) -> {
+                listener.success(new IdentificationSegment(start, end));
+            });
+
+            if (updateSuccess) {
+                return;
+            } else {
+                try {
+                    Thread.sleep(new Random().nextInt(1000));
+                } catch (InterruptedException e) {
+                }
+            }
+        }
+
+        listener.failed();
+    }
+
+    public void fetchSegmentInBackGround(final IdentificationCache cache) {
+        if (task != null) {
+            logger.info("Fetch segment task already running.");
+            return;
+        }
+
+        task = Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(() -> fetchSegment(new Listener() {
```

Comment:
Create a thread task for without any lock. When your cache fails, maybe more than one request ask for `fetchSegmentInBackGround`, the `task != null` can't guarantee the task is initialized once.